src/diffusers/models/attention.py

```diff
@@ -187,95 +187,137 @@ def __init__(
         dim: int,
         num_attention_heads: int,
         attention_head_dim: int,
-        dropout=0.0,
+        dropout: Optional[float] = 0.0, 
         cross_attention_dim: Optional[int] = None,
         activation_fn: str = "geglu",
         num_embeds_ada_norm: Optional[int] = None,
         attention_bias: bool = False,
-        only_cross_attention: bool = False,
```

Comment:
Note we cannot remove existing config parameters because it would break existing models. Could we maybe leave this one and remove `attn1_type` ? :-)